config/initializers/doorkeeper_openid_connect.rb

```diff
@@ -49,32 +49,34 @@ def @config.signing_key
   # expiration 600
 
   claims do
-    claim :email, &:email
+    claim :email, scope: :openid do |resource_owner, _scopes, _access_token|
+      # Pass the resource_owner's preferred_username if the application has
+      # `profile` scope access. Otherwise, provide a more generic alternative.
+      # scopes.exists?(:profile) ? resource_owner.username : "summer-sun-9449"
+      resource_owner.email
+    end
 
     claim :email_verified do |_resource_owner|
       true
     end
 
-    claim :groups do |resource_owner|
-      resource_owner.asserted_groups || []
+    claim :groups, scope: :openid do |resource_owner|
+      (resource_owner.asserted_groups || []).map(&:name)
     end
 
-    claim :name do |resource_owner|
+    claim :name, scope: :openid do |resource_owner|
       # read_name is aliased to name in the User model, and is called real_name
       # here because of a doorkeeper issue where calling `.name` ends up
       # raising an exception.
       resource_owner.try(:real_name) or resource_owner.username
     end
 
-    claim :nickname, &:username
-
-    claim :preferred_username, &:username
-
-    # claim :preferred_username, scope: :openid do |resource_owner, scopes, access_token|
-    #   # Pass the resource_owner's preferred_username if the application has
-    #   # `profile` scope access. Otherwise, provide a more generic alternative.
-    #   scopes.exists?(:profile) ? resource_owner.username : "summer-sun-9449"
-    # end
+    claim :username, scope: :openid do |resource_owner, _scopes, _access_token|
+      # Pass the resource_owner's preferred_username if the application has
```

Comment:
the way I'm reading this documentation that `preferred_username` can be overwritten, so if the documentation is correct, then I'm interpreting the code as: pass along the *preferred* username

Reply:
ah, I missed renaming the comment as well. Changing usernames is _potentially_ possible if the admins enable changing usernames and emails.